### PR TITLE
cdhit: new package

### PR DESCRIPTION
**Pull request by akthoma**
adding new package cdhit

### CODE REVIEW

File: var/spack/repos/builtin/packages/cdhit/package.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+    def build(self, spec, prefix):
+        mkdirp(prefix.bin)
+        if '~openmp' in spec:
+            make('openmp=no')
+        else:
+            make()
+
+    def setup_environment(self, spack_env, run_env):
+        spack_env.set('PREFIX', prefix.bin)
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/cdhit/package.py]: Is this variable needed by the application to run correctly?

Comment (akthoma) [inline var/spack/repos/builtin/packages/cdhit/package.py]: Not necessarily. By default, the openmp variable is true but is able to be switched off for older systems that don't offer openmp multi-threaded support, per the README:

> For cd-hit
> 
> How to compile?
>   1. Compile with multi-threading support (default): make
>   2. Compile without multi-threading support (if you are on very old systems): make openmp=no

Comment (adamjstewart) [inline var/spack/repos/builtin/packages/cdhit/package.py]: @alalazo This is MakefilePackage. I assume the Makefile has a hard-coded PREFIX. It's setting `spack_env`, not `run_env`, so it isn't for the module file.

Comment (alalazo) [inline var/spack/repos/builtin/packages/cdhit/package.py]: >  It's setting spack_env, not run_env, so it isn't for the module file.

:+1:

Comment (akthoma) [inline var/spack/repos/builtin/packages/cdhit/package.py]: @alalazo my apologies... I was looking at the wrong line 🤦‍♀️ . But yes to what @adamjstewart said!

Comment (alalazo) [inline var/spack/repos/builtin/packages/cdhit/package.py]: @akthoma No worries!

**Pull request merged**